### ISSUE TITLE
Possible bug: panderOptions('round', 2) being lost when dplyr::group_by used

### ISSUE THREAD

**Issue by brfry** [Rapporter/pander#287]
Not sure if it is something I am overlooking, a problem with dplyr, or something with pander, but I am having this issue:

Load libraries and set options:

```{r}
library(pander)
panderOptions('round', 2)
panderOptions('keep.trailing.zeros', TRUE)
panderOptions('table.split.table', Inf)

library(tidyverse)
```

Create some data:

```{r}
set.seed(10)
df <- data.frame(x = rnorm(10), y = rnorm(10), class = c("a", "b"))

print(df)
```

  x | y | class
---|---|---
  0.01874617	| 1.10177950	| a		
  -0.18425254	| 0.75578151	| b		
  -1.37133055	|-0.23823356	|a		
-0.59916772	|0.98744470	|b		
0.29454513	|0.74139013	|a		
0.38979430	|0.08934727	|b		
-1.20807618	|-0.95494386	|a		
-0.36367602	|-0.19515038	|b		
-1.62667268	|0.92552126	|a		
-0.25647839	|0.48297852	|b

Manipulate df and use pander without any group_by() operation:

```{r}
# make a table and output data
df_nogroup <- df %>%
  mutate(xy = x * y) %>%
  summarise(mean = mean(xy, na.rm = TRUE),
            sd = sd(xy, na.rm = TRUE),
            se = sd(xy, na.rm = TRUE)/sqrt(n()),
            CI95_upr = mean + (qnorm(0.975) * se),
            CI95_lwr = mean - (qnorm(0.975) * se),
            n = n())
  
pander(df_nogroup, "No grouping step. Round working")

```

------------------------------------------
 mean |  sd  | se |  CI95_upr |  CI95_lwr  | n 
------| ---- |----| ---------- |----------| ---
-0.05 | 0.68| 0.21  |  0.37   |   -0.47   | 10 
------------------------------------------

Table: No grouping step. Round working

Now with group_by():
```{r}
df_group <- df %>%
  mutate(xy = x * y) %>%
  group_by(class) %>%
  summarise(mean = mean(xy, na.rm = TRUE),
            sd = sd(xy, na.rm = TRUE),
            se = sd(xy, na.rm = TRUE)/sqrt(n()),
            CI95_upr = mean + (qnorm(0.975) * se), 
            CI95_lwr = mean - (qnorm(0.975) * se),
            n = n()) 

pander(df_group, "Grouping appears to be the culprit")

```

--------------------------------------------------------
 class  |  mean  |   sd   |  se  |  CI95_upr  | CI95_lwr|   n 
------- |--------| ------ |------| ---------- |----------| ---
   a    |0.04277  |0.9674 |0.4326 | 0.89069  |  -0.8051  |  5 
   b  |  -0.14979| 0.2640| 0.1181|  0.08163 |   -0.3812 |   5 
--------------------------------------------------------

Table: Grouping appears to be the culprit

I tried adding a ungroup() to the end of the piping with no change.

**Comment by daroczig** [comment]
Cannot reproduce the bug here -- can you please share your `devtools::session_info()`?

**Comment by daroczig** [comment]
Can you try the dev version of `pander`? It hasn't been updated on CRAN for a while, and I think this was fixed since then at #234. Based on this, I'm closing this now, but please feel free to reopen if that doesn't solve the issue.

**Comment by brfry** [comment]
That has fixed it. Works with both panderOptions global and inline. Thanks.
